src/components/error-component/ErrorBoundary.js

```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+
+class ErrorBoundary extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { hasError: false };
+  }
+
+  static getDerivedStateFromError(error) {
+    return { hasError: true };
+  }
+
+  componentDidCatch(error, info) {
+    // Testing Error Blocks Manually
+    // this.setState({ hasError: true });
```

Comment:
One of the goals of the task is to catch error. Any specific reason for not having the code in place for catching the error?